kratos/includes/user_provided_linear_master_slave_constraint.h

```diff
@@ -0,0 +1,322 @@
+     * creates a new constraint pointer
+     * @param NewId the ID of the new element
+     * @param ThisNodes the nodes of the new element
+     * @param pProperties the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    MasterSlaveConstraint::Pointer Create(IndexType Id,
+                                                DofPointerVectorType& MasterDofsVector,
+                                                DofPointerVectorType& SlaveDofsVector,
+                                                const MatrixType& RelationMatrix,
+                                                const VectorType& ConstantVector) const override
+    {
+        KRATOS_TRY
+        auto new_pointer = Kratos::make_shared<UserProvidedLinearMasterSlaveConstraint>(Id, MasterDofsVector, SlaveDofsVector, RelationMatrix, ConstantVector);
+        return new_pointer;
+        KRATOS_CATCH("");
+    }
+
+    MasterSlaveConstraint::Pointer Create(IndexType Id,
+                                        NodeType& rMasterNode,
+                                        const VariableType& rMasterVariable,
+                                        NodeType& rSlaveNode,
+                                        const VariableType& rSlaveVariable,
+                                        const double Weight,
+                                        const double Constant) const override
+    {
+        KRATOS_TRY
+        return Kratos::make_shared<UserProvidedLinearMasterSlaveConstraint>(Id, rMasterNode, rMasterVariable, rSlaveNode, rSlaveVariable, Weight, Constant);
+        KRATOS_CATCH("");
+    }
+
+
+    ///@}
+
+    ///@name Access
+    ///@{
+
+    /**
+     * this determines the master equation IDs connected to this constraint
+     * @param rResult the elemental equation ID vector
+     * @param rCurrentProcessInfo the curent process info instance
+     */
+    void EquationIdVector(EquationIdVectorType& rSlaveEquationIds,
+                                  EquationIdVectorType& rMasterEquationIds,
+                                  ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (rSlaveEquationIds.size() != mSlaveDofsVector.size())
+            rSlaveEquationIds.resize(mSlaveDofsVector.size());
+
+        if (rMasterEquationIds.size() != mMasterDofsVector.size())
+            rMasterEquationIds.resize(mMasterDofsVector.size());
+
+        for(unsigned int i=0; i<rSlaveEquationIds.size(); ++i)
+            rSlaveEquationIds[i] = mSlaveDofsVector[i]->EquationId();
+
+        for(unsigned int i=0; i<rMasterEquationIds.size(); ++i)
+            rMasterEquationIds[i] = mMasterDofsVector[i]->EquationId();
+    }
+
+    /**
+     * this is called during the assembling process in order
+     * to calculate all elemental contributions to the global system
+     * matrix and the right hand side
+     * @param rTransformationMatrix the elemental left hand side matrix
+     * @param rConstant the elemental right hand side
```

Comment:
It's better if we don't comment overrides. The comment can be found in the base method and we avoid diverging / wrong comments.